kong/plugins/ai-proxy/schema.lua

```diff
@@ -0,0 +1,146 @@
+local typedefs = require "kong.db.schema.typedefs"
+
+local request_log_compatible = {
+  "llm/v1/chat",
+  "llm/v1/completions",
+}
+
+local auth_schema = {
+  type = "record",
+  required = false,
+  fields = {
+    { header_name = { type = "string", description = "If AI model requires authentication via Authorization or API key header, specify its name here.", required = false, referenceable = true } },
+    { header_value = { type = "string", description = "Specify the full auth header value for 'header_name', for example 'Bearer key' or just 'key'.", required = false, referenceable = true } },
+    { param_name = { type = "string", description = "If AI model requires authentication via query parameter, specify its name here.", required = false, referenceable = true } },
+    { param_value = { type = "string", description = "Specify the full parameter value for 'param_name'.", required = false, referenceable = true } },
+    { param_location = { type = "string", description = "Specify whether the 'param_name' and 'param_value' options go in a query string, or the POST form/JSON body.", required = false, one_of = { "query", "body" } } },
```

Comment:
suggest asking the user to input if the request body is a form or json

Reply:
It's only like this because it can't be both `form` and `json` at the same time, so I just guessed to combine the two together.

Reply:
Is it possible to know at configuration time that it will be json or form? If yes, asking the user is a bit more predictable. If both (some requests have a json body while others have a form body) are possible then we can't do much.
requesting this change to make the behavior more explicit

Reply:
Honestly I don't think we will EVER need to have it as JSON or FORM param... this seems like it wouldn't happen.

Maybe "cookie" might need support one day, but I can just remove "body" as an option to make it simpler for first release?